### PR TITLE
New sample: Raster Function Service

### PR DESCRIPTION
**Pull request by dkemlage**

### CODE REVIEW

**Comment by dkemlage** [comment]
@nCastle1 Please code review the new sample: Raster Function Service. I am available for questions if you have any.

File: src/UWP/ArcGISRuntime.UWP.Samples/ArcGISRuntime.UWP.Samples.VBNet.vbproj [hunk 1]
```diff
@@ -109,6 +110,8 @@
     </Content>
     <None Include="Samples\Layers\RasterLayerImageServiceRaster\metadata.json" />
     <None Include="Samples\Layers\RasterLayerImageServiceRaster\README.md" />
+    <None Include="Samples\Layers\RasterLayerRasterFunction\metadata.json" />
+    <None Include="Samples\Layers\RasterLayerRasterFunction\README.md" />
```
Comment (nCastle1) [inline src/UWP/ArcGISRuntime.UWP.Samples/ArcGISRuntime.UWP.Samples.VBNet.vbproj]: I don't think we want to include the readme.md files in the solutions/projects. It looks like the readme was included for the RasterLayerImageServiceRaster sample as well, so I don't know how you want to handle that.

File: src/WPF/ArcGISRuntime.WPF.Samples/ArcGISRuntime.WPF.Samples.VBNet.vbproj [hunk 2]
```diff
@@ -191,6 +193,8 @@
     <None Include="Samples\Layers\FeatureLayerUrl\metadata.json" />
     <None Include="Samples\Layers\RasterLayerImageServiceRaster\metadata.json" />
     <None Include="Samples\Layers\RasterLayerImageServiceRaster\README.md" />
+    <None Include="Samples\Layers\RasterLayerRasterFunction\metadata.json" />
+    <None Include="Samples\Layers\RasterLayerRasterFunction\README.md" />
```
Comment (nCastle1) [inline src/WPF/ArcGISRuntime.WPF.Samples/ArcGISRuntime.WPF.Samples.VBNet.vbproj]: More readmes included; I suspect there are readmes in other samples as well. Maybe there should be an issue to remove them?

**Comment by nCastle1** [comment]
It might be more straightforward to use verbatim strings for the JSON string literals. I tested and it works in both C# and VB:

```
String jsonString = @"{
                ""raster_function_arguments"":
              {
                    ""z_factor"":{ ""double"":25.0,""type"":""Raster_function_variable""},
                ""slope_type"":{ ""raster_slope_type"":""none"",""type"":""Raster_function_variable""},
                ""azimuth"":{ ""double"":315,""type"":""Raster_function_variable""},
                ""altitude"":{ ""double"":45,""type"":""Raster_function_variable""},
                ""type"":""Raster_function_arguments"",
                ""raster"":{ ""name"":""raster"",""is_raster"":true,""type"":""Raster_function_variable""},
                ""nbits"":{ ""int"":8,""type"":""Raster_function_variable""}
                },
              ""raster_function"":{ ""type"":""Hillshade_function""},
              ""type"":""Raster_function_template""
            }";
```

and VB:

```
Dim jsonString As String = "{
                ""raster_function_arguments"":
              {
                    ""z_factor"":{ ""double"":25.0,""type"":""Raster_function_variable""},
                ""slope_type"":{ ""raster_slope_type"":""none"",""type"":""Raster_function_variable""},
                ""azimuth"":{ ""double"":315,""type"":""Raster_function_variable""},
                ""altitude"":{ ""double"":45,""type"":""Raster_function_variable""},
                ""type"":""Raster_function_arguments"",
                ""raster"":{ ""name"":""raster"",""is_raster"":true,""type"":""Raster_function_variable""},
                ""nbits"":{ ""int"":8,""type"":""Raster_function_variable""}
                },
              ""raster_function"":{ ""type"":""Hillshade_function""},
              ""type"":""Raster_function_template""
            }"
```

File: src/WPF/ArcGISRuntime.WPF.Samples/ArcGISRuntime.WPF.Samples.VBNet.vbproj [hunk 3]
```diff
@@ -89,6 +89,8 @@
     </Content>
     <Content Include="Samples\GraphicsOverlay\SketchOnMap\SketchOnMap.jpg" />
     <Content Include="Samples\Layers\RasterLayerImageServiceRaster\RasterLayerImageServiceRaster.jpg" />
+    <Content Include="Samples\Layers\RasterLayerRasterFunction\RasterLayerRasterFunction.jpg" />
+    <Content Include="Samples\Layers\RasterLayerRasterFunction\RasterLayerRasterFunction.xaml.cs" />
```
Comment (nCastle1) [inline src/WPF/ArcGISRuntime.WPF.Samples/ArcGISRuntime.WPF.Samples.VBNet.vbproj]: It looks like the C# versions of the files were included in the VB project.

**Comment by nCastle1** [comment]
I would recommend changing all of the URLs to https. I wouldn't be surprised if Android or UWP started enforcing something similar to ATS in the future and the change will save us some future effort if that happens.

**Comment by nCastle1** [comment]
Aside from the noted changed and suggestions, this looks like a great sample. Pending changes, you're good to merge.

**Comment by dkemlage** [comment]
Made all changes as per code review.

**Pull request merged**